modules/client/src/test/java/org/apache/ignite/client/ClientKeyValueBinaryViewTest.java

```diff
@@ -330,4 +331,96 @@ public void testGetAndPut() {
         assertEquals("1", res2.stringValue(0));
         assertEquals("3", kvView.get(null, tupleKey(1L)).stringValue(0));
     }
+
+    @Test
+    public void testGetNullable() {
+        KeyValueView<Tuple, Tuple> kvView = defaultTable().keyValueView();
+
+        Tuple existingKey = tupleKey(DEFAULT_ID);
+        Tuple nonExistingKey = tupleKey(-1L);
+
+        kvView.put(null, existingKey, null);
+        kvView.remove(null, nonExistingKey);
+
+        NullableValue<Tuple> nullVal = kvView.getNullable(null, existingKey);
+        NullableValue<Tuple> missingVal = kvView.getNullable(null, nonExistingKey);
+
+        assertNull(missingVal);
+
+        assertNotNull(nullVal);
+        assertNotNull(nullVal.get());
+        assertEquals(DEFAULT_ID, nullVal.get().longValue("id"));
```

Comment:
Do we really expect the value contains key field?

Reply:
This is consistent with `get()` operation for binary view
```
        Tuple existingKey = tupleKey(DEFAULT_ID);
        kvView.put(null, existingKey, null);
        Tuple nullVal = kvView.get(null, existingKey);

        assertEquals(DEFAULT_ID, nullVal.longValue("id"));
```

Or do you mean I need to remove this explicit check?

Reply:
:thinking:  but this is not consistent with embedded binary view behavior

Are we expecting only value columns in tuple?
If so - I can rework tests and file a ticket to fix this.
Or I can try to fix it here :thinking:

Reply:
No, it's a bug,
filed separate ticket for this https://issues.apache.org/jira/browse/IGNITE-23164
because I can't come up with a quick solution to this problem